gluten-core/src/main/scala/org/apache/spark/sql/execution/ColumnarCollapseTransformStages.scala

```diff
@@ -150,12 +150,6 @@ case class ColumnarCollapseTransformStages(
   /** Inserts an InputIteratorTransformer on top of those that do not support transform. */
   private def insertInputIteratorTransformer(plan: SparkPlan): SparkPlan = {
     plan match {
-      case p: CartesianProductExecTransformer =>
-        p.withNewChildren(
-          p.children.map(
-            child =>
-              ColumnarCollapseTransformStages.wrapInputIteratorTransformer(
-                insertWholeStageTransformer(child))))
```

Comment:
We should avoid changing the core code for dedicated operator which makes the code hard to read and maintain.